test/strings/basic.jl

```diff
@@ -938,6 +938,17 @@ end
     end
 end
 
+@testset "Conversion to Type{Union{String, SubString{String}}}" begin
+    str::String = "abc"
```

Comment:
This `::String` seems redundant?

Reply:
I wrote it just to be sure and also as a way of documenting